Code/ForceField/MMFF/Params.cpp

```diff
@@ -28,14 +28,13 @@ typedef boost::tokenizer<boost::char_separator<char> > tokenizer;
 namespace ForceFields {
 namespace MMFF {
 
-class std::unique_ptr<MMFFAromCollection> MMFFAromCollection::ds_instance = nullptr;
-
 extern const std::uint8_t defaultMMFFArom[];
 
 MMFFAromCollection *MMFFAromCollection::getMMFFArom(
     const std::uint8_t *mmffArom) {
-  if (!ds_instance || mmffArom) {
-    ds_instance.reset(new MMFFAromCollection(mmffArom));
+  static std::unique_ptr<MMFFAromCollection> ds_instance(new MMFFAromCollection(nullptr));
+  if(mmffArom){
+    return new MMFFAromCollection(mmffArom);
```

Comment:
Won't this leak like a sieve?  The caller can't delete this as one is a unique_ptr and one is not.

Reply:
I will add something to the documentation here indicating when the results should be delete'd
And, yes, it will leak in those rare cases where the user provides their own MMFF parameters.

Reply:
hmm, the semantics of that changed too.

Reply:
Ugly

Reply:
The same code would also work, static initialization is thread safe in c++11 and is a bit simpler.

```
MMFFAromCollection *MMFFAromCollection::getMMFFArom(
    const std::uint8_t *mmffArom) {
 static MMFFAromCollection df_instance(nullptr);
 if(mmffAtom) {
   return new MMFFAromCollection(mmffArom);
 }
 return &df_instance;
}
```